nanek/pn-logging
# Chore 177333094 upgrade winston

### What
This upgrades pn-logging to use winston version 3 and the winston-loggly-bulk library for sending logs to loggly.

I followed the notes for the Winston 3: https://github.com/winstonjs/winston/blob/master/UPGRADE-3.0.md

One of the main changes was that the formats are now functions, so I built out a function to parse our options and apply the correct format.

### Test
1. In sd-auth, set the package.json file to use the hash of this branch `434627d504fd881cb0db66669c7ed5466d2eb58c` for pn-logging.
2. Run yarn
3. Update local-development to use logging for both console and loggly:

```
logging: {
    transports: [
      {
        Console: {
          level: 'info',
          json: true,
          prettyPrint: true,
          colorize: true,
        },
      },
      {
        Loggly: {
          level: 'info',
          json: true,
          tag: 'sd-auth',
          subdomain: '',
          token: '',
        },
      },
],}
```
4. Look up the subdomain and and token using sd-ecs for sd-auth
5. Rebuild sd-auth and start the service
6. Go to sdlocal.com and log-in and out
7. Inspect the console to ensure everything is being printed correctly
8. Go to loggly and look up your logs by serverHost
9. Ensure the loggly logs look correct
10. Play around with the console settings (prettyPrint, colorize) and ensure they are working correctly
